src/Orchard.Web/Modules/Orchard.MessageBus/Services/SqlServerServiceBrokerFeatureGuard.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Web.Mvc;
+using Orchard.Environment;
+using Orchard.Environment.Configuration;
+using Orchard.Environment.Extensions.Models;
+using Orchard.Environment.Features;
+using Orchard.Localization;
+using Orchard.Mvc;
+using Orchard.Mvc.Filters;
+using Orchard.UI.Notify;
+
+namespace Orchard.MessageBus.Services
+{
+    /// <summary>
+    /// Prevents the SQL Server Service Broker feature from being enabled for tenants that are not using
+    /// Microsoft SQL Server as their database provider.
+    /// </summary>
+    /// <remarks>
+    /// The implementation is hackish but there seems to be no other way: if it would use <see cref="Orchard.UI.Notify.INotifier"/>
+    /// it wouldn't work since <see cref="Orchard.UI.Notify.NotifyFilter"/> that writes out notifications to TempData 
+    /// runs before feature events are raised. Thus we have to manually add the messages to TempData, just as the filter 
+    /// would do.
+    /// </remarks>
+    public class SqlServerServiceBrokerFeatureGuard : FilterProvider, IFeatureEventHandler, IActionFilter
+    {
+        private const string FeatureId = "Orchard.MessageBus.SqlServerServiceBroker";
+        private const string TempDataKey = FeatureId + ".TempData";
+
+        private readonly ShellSettings _shellSettings;
+        private readonly IFeatureManager _featureManager;
+        private readonly IHttpContextAccessor _hca;
+
+        public Localizer T { get; set; }
+
+        public SqlServerServiceBrokerFeatureGuard(
+            ShellSettings shellSettings,
+            IFeatureManager featureManager,
+            IHttpContextAccessor hca)
+        {
+            _shellSettings = shellSettings;
+            _featureManager = featureManager;
+            _hca = hca;
+
+            T = NullLocalizer.Instance;
+        }
+
+        void IActionFilter.OnActionExecuting(ActionExecutingContext filterContext) { }
+
+        void IActionFilter.OnActionExecuted(ActionExecutedContext filterContext)
+        {
+            var httpContext = _hca.Current();
+
+            if (httpContext == null) return;
+
+            httpContext.Items[TempDataKey] = filterContext.Controller.TempData;
+        }
+
+        public void Installing(Feature feature) { }
+
+        public void Installed(Feature feature) { }
+
+        public void Enabling(Feature feature) { }
```

Comment:
Please check if we can store here whether `SqlServerServiceBroker`'s dependencies (there's only `Orchard.MessageBus`, but don't wire that in) are already enabled or not. Those that aren't should also be disabled along with `SqlServerServiceBroker` in the `Enabled` handler.